File: 611-valid-triangle-number/valid-triangle-number.py
class Solution:
    def triangleNumber(self, nums: list[int]) -> int:
        nums.sort()
        count = 0
        for i in range(len(nums) - 1, -1, -1):
            l = 0
            r = i - 1
            while l < r:
                if nums[l] + nums[r] > nums[i]:
                    count += (r - l)
                    r -= 1
                elif nums[l] + nums[r] <= nums[i]:
                    l += 1
        return count

File: 611-valid-triangle-number/valid-triangle-number.c
int compare(const void *a, const void *b) {
    return (*(int*)a - *(int*)b);
}
int triangleNumber(int* nums, int numsSize) {
    qsort(nums, numsSize, sizeof(int), compare);
    int count = 0;
    for (int i = numsSize - 1; i >= 2; i--) {
        int l = 0;
        int r = i - 1;
        while (l < r) {
            if (nums[l] + nums[r] > nums[i]) {
                count += (r - l);
                r--;
            }
            else {
                l++;
            }
        }
    }
    return count;
}